compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -2045,7 +2044,7 @@ pub(super) fn check_type_bounds<'tcx>(
         .collect();
```

Comment:
Can we remove the `collect` here? It doesn't look like it's driving any side effects, is the `debug!` below important?